Make mainFrame() return an AbstractFrame

chirags27 opened the pull request:
#### 52ec37994c2d2ff334b0e6ee531ac662872e231e
<pre>
Make mainFrame() return an AbstractFrame
<a href="https://bugs.webkit.org/show_bug.cgi?id=252186">https://bugs.webkit.org/show_bug.cgi?id=252186</a>
rdar://105412703

Reviewed by Alex Christensen.

In preparation for website isolation, lets make Frame::mainFrame()
return an AbstractFrame. Going forward, the callers of this method will
need to decide if they are dealing with a LocalFrame or not, and take
action accordingly.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::mainFrame const):
* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
(WebCore::ContentExtensions::ContentExtensionsBackend::processContentRuleListsForLoad):
* Source/WebCore/css/query/MediaQueryFeatures.cpp:
(WebCore::MQ::Features::color):
(WebCore::MQ::Features::colorGamut):
(WebCore::MQ::Features::deviceAspectRatio):
(WebCore::MQ::Features::deviceHeight):
(WebCore::MQ::Features::deviceWidth):
(WebCore::MQ::Features::dynamicRange):
(WebCore::MQ::Features::monochrome):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::resolveStyle):
(WebCore::Document::topDocument const):
(WebCore::computeIntersectionState):
* Source/WebCore/dom/DocumentMarkerController.cpp:
(WebCore::updateMainFrameLayoutIfNeeded):
* Source/WebCore/dom/EventDispatcher.cpp:
(WebCore::shouldSuppressEventDispatchInDOM):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::canPaste const):
(WebCore::Editor::shouldInsertText const):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::allowExecutionWhenDisabledPaste):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::mainDocumentRegistrableDomainForPCM const):
(WebCore::HTMLAnchorElement::parsePrivateClickMeasurement const):
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::isElementMainContentForPurposesOfAutoplay):
(WebCore::isElementRectMostlyInMainFrame):
(WebCore::isElementLargeRelativeToMainFrame):
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::didPaint):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::disallowWebArchive const):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::userAgent const):
(WebCore::FrameLoader::navigatorPlatform const):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
(WebCore::FrameLoader::loadResourceSynchronously):
(WebCore::FrameLoader::continueLoadAfterNavigationPolicy):
(WebCore::FrameLoader::loadDifferentDocumentItem):
(WebCore::FrameLoader::loadProgressingStatusChanged):
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::FrameLoader::HistoryController::updateForCommit):
(WebCore::FrameLoader::HistoryController::updateForSameDocumentNavigation):
(WebCore::FrameLoader::HistoryController::updateBackForwardListClippedAtTarget):
(WebCore::FrameLoader::HistoryController::pushState):
* Source/WebCore/loader/NavigationDisabler.h:
(WebCore::NavigationDisabler::NavigationDisabler):
(WebCore::NavigationDisabler::~NavigationDisabler):
(WebCore::NavigationDisabler::isNavigationAllowed):
* Source/WebCore/loader/cache/CachedResource.cpp:
(WebCore::CachedResource::load):
* Source/WebCore/loader/cache/CachedResource.h:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::CachedResourceLoader::requestResource):
* Source/WebCore/page/AutoscrollController.cpp:
(WebCore::AutoscrollController::stopAutoscrollTimer):
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::outerHeight const):
(WebCore::DOMWindow::outerWidth const):
(WebCore::DOMWindow::isSameSecurityOriginAsMainFrame const):
(WebCore::DOMWindow::open):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::hitTestResultAtPoint const):
(WebCore::EventHandler::selectCursor):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::internalKeyEvent):
* Source/WebCore/page/Frame.h:
(WebCore::Frame::mainFrame const):
* Source/WebCore/page/FrameTree.cpp:
(WebCore::isFrameFamiliarWith):
(WebCore::FrameTree::find const):
(WebCore::FrameTree::traverseNext const):
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::axObjectCache const):
* Source/WebCore/page/InteractionRegion.cpp:
(WebCore::interactionRegionForRenderedRegion):
* Source/WebCore/page/IntersectionObserver.cpp:
(WebCore::IntersectionObserver::IntersectionObserver):
* Source/WebCore/page/Screen.cpp:
(WebCore::isLoadingInHeadlessMode):
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::mainDocumentLoader):
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
(WebCore::ScrollingCoordinator::updateSynchronousScrollingReasons):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::scrollBySimulatingWheelEvent):
* Source/WebCore/testing/Internals.mm:
(WebCore::Internals::rangeForDictionaryLookupAtLocation):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):
(WebKit::WebAutomationSessionProxy::takeScreenshot):
(WebKit::WebAutomationSessionProxy::snapshotRectForScreenshot):
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::WebLoaderStrategy::loadResource):
(WebKit::addParametersShared):
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect):
* Source/WebKit/WebProcess/WebCoreSupport/ShareableBitmapUtilities.cpp:
(WebKit::createShareableBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp:
(WebKit::WebResourceLoadObserver::logSubresourceLoading):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::convertDragImageToBitmap):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::isTopFrameNavigatingToAppBoundDomain const):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::rangeForPointInRootViewCoordinates):
* Source/WebKitLegacy/mac/WebView/WebImmediateActionController.mm:
(-[WebImmediateActionController performHitTestAtPoint:]):
* Source/WebKitLegacy/win/WebCoreSupport/AcceleratedCompositingContext.cpp:
(AcceleratedCompositingContext::flushPendingLayerChanges):
* Source/WebKitLegacy/win/WebCoreSupport/WebDragClient.cpp:
(WebDragClient::dragSourceActionMaskForPoint):

Canonical link: <a href="https://commits.webkit.org/260414@main">https://commits.webkit.org/260414@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/504b16f4b44db9fed0741429cc8ae76416701f32

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108215 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17300 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41082 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117338 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116646 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112102 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18678 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8584 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100423 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113983 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97285 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/42000 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28927 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83678 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10154 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30272 "Passed tests") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/38/builds/96775 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10886 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7177 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/38/builds/96775 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16303 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49867 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7203 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12468 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
```diff
@@ -211,8 +211,11 @@ ContentRuleListResults ContentExtensionsBackend::processContentRuleListsForLoad(
-        else if (auto* mainDocument = frame->mainFrame().document())
-            mainDocumentURL = mainDocument->url();
+        else if (auto* localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame())) {
+            auto* mainDocument = localFrame->document();
```
This can be in the if statement below.

achristensen07 commented on Source/WebCore/css/query/MediaQueryFeatures.cpp:
```diff
@@ -248,7 +248,9 @@ const FeatureSchema& color()
-            return screenDepthPerComponent(context.document.frame()->mainFrame().view());
+            if (auto* localFrame = dynamicDowncast<LocalFrame>(context.document.frame()->mainFrame()))
+                return screenDepthPerComponent(localFrame->view()); 
+            return -1;
```
Let's return 8 for now, which is the default value on iOS.

achristensen07 commented on Source/WebCore/css/query/MediaQueryFeatures.cpp:
```diff
@@ -264,8 +266,10 @@ const FeatureSchema& colorGamut()
-            if (screenSupportsExtendedColor(frame.mainFrame().view()))
-                identifiers.append(CSSValueP3);
+            if (auto* localFrame = dynamicDowncast<LocalFrame>(frame.mainFrame())) {
+                if (localFrame->view())
```
You dropped the screenSupportsExtendedColor call.

achristensen07 commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -2176,7 +2176,7 @@ void Document::resolveStyle(ResolveStyleType type)
         // to check if any other elements ended up under the mouse pointer due to re-layout.
         if (m_hoveredElement && !m_hoveredElement->renderer() && is<LocalFrame>(frameView.frame()))
-            downcast<LocalFrame>(frameView.frame()).mainFrame().eventHandler().dispatchFakeMouseMoveEventSoon();
+            downcast<LocalFrame>(downcast<LocalFrame>(frameView.frame()).mainFrame()).eventHandler().dispatchFakeMouseMoveEventSoon();
```
Let's add dynamicDowncast and null checks here too.

achristensen07 commented on Source/WebCore/html/HTMLAnchorElement.cpp:
```diff
@@ -525,7 +530,9 @@ std::optional<PrivateClickMeasurement> HTMLAnchorElement::parsePrivateClickMeasu
     RegistrableDomain mainDocumentRegistrableDomain;
-    if (auto mainDocument = frame->mainFrame().document())
+    auto* localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
+    ASSERT(localFrame);
```
Let's just do a null check instead of an assert.

achristensen07 commented on Source/WebCore/html/MediaElementSession.cpp:
```diff
@@ -992,7 +992,10 @@ static bool isElementMainContentForPurposesOfAutoplay(const HTMLMediaElement& el
 
-    auto& mainFrame = document.frame()->mainFrame();
+    auto* localFrame = dynamicDowncast<LocalFrame>(document.frame()->mainFrame());
+    ASSERT(localFrame);
```
early return if this is null.

achristensen07 commented on Source/WebCore/html/MediaElementSession.cpp:
```diff
@@ -1054,10 +1061,13 @@ static bool isElementLargeRelativeToMainFrame(const HTMLMediaElement& element)
 
-    if (!documentFrame->mainFrame().view())
+    auto* localFrame = dynamicDowncast<LocalFrame>(documentFrame->mainFrame());
+    ASSERT(localFrame);
```
Early return if null.

achristensen07 commented on Source/WebCore/loader/DocumentLoader.cpp:
```diff
@@ -1068,7 +1068,10 @@ bool DocumentLoader::disallowWebArchive() const
     // On purpose of maintaining existing tests.
-    if (frame()->mainFrame().loader().alwaysAllowLocalWebarchive())
+    auto* localFrame = dynamicDowncast<LocalFrame>(frame()->mainFrame());
+    ASSERT(localFrame);
```
Early return if false.

achristensen07 commented on Source/WebCore/loader/NavigationDisabler.h:
```diff
@@ -35,16 +35,17 @@ class NavigationDisabler {
     {
-        if (frame)
-            ++frame->mainFrame().m_navigationDisableCount;
+        auto* localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
```
We need to null check frame before this dereference.

achristensen07 commented on Source/WebCore/loader/cache/CachedResource.cpp:
```diff
@@ -184,7 +184,17 @@ void CachedResource::load(CachedResourceLoader& cachedResourceLoader)
-    if (auto* topDocument = frame.mainFrame().document()) {
+    if (!dynamicDowncast<LocalFrame>(frame.mainFrame())) {
+        // FIXME: Fix with website isolation work
+        failBeforeStarting();
```
Let's just skip the use of topDocument if the main frame isn't local, instead of failing all loads or early returning.

achristensen07 commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -1209,7 +1209,8 @@ HitTestResult EventHandler::hitTestResultAtPoint(const LayoutPoint& point, Optio
     // otherwise we might hit areas that are obscured by higher frames.
     if (!m_frame.isMainFrame()) {
-        Ref mainFrame = m_frame.mainFrame();
+        ASSERT(dynamicDowncast<LocalFrame>(m_frame.mainFrame()));
```
Let's skip this code if it's not local.

achristensen07 commented on Source/WebCore/page/FrameView.cpp:
```diff
@@ -5719,7 +5719,9 @@ AXObjectCache* FrameView::axObjectCache() const
     if (!cache && !m_frame->isMainFrame()) {
-        if (auto* mainFrameDocument = m_frame->mainFrame().document())
+        auto localMainFrame = dynamicDowncast<LocalFrame>(m_frame->mainFrame());
+        ASSERT(localMainFrame);
```
Let's null check this in the if statement below.

achristensen07 commented on Source/WebCore/page/IntersectionObserver.cpp:
```diff
@@ -130,8 +130,11 @@ IntersectionObserver::IntersectionObserver(Document& document, Ref<IntersectionO
-    } else if (auto* frame = document.frame())
-        m_implicitRootDocument = frame->mainFrame().document();
+    } else if (auto* frame = document.frame()) {
+        auto* localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
```
This can be in the if statement below:
if (auto* localFrame = ...)

achristensen07 commented on Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
```diff
@@ -982,7 +989,15 @@ void WebAutomationSessionProxy::snapshotRectForScreenshot(WebCore::PageIdentifie
+        completionHandler(internalErrorType, { });
+        return;
+    }
+
```
extra space.

achristensen07 commented on Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
```diff
@@ -312,7 +315,10 @@ static void addParametersShared(const Frame* frame, NetworkResourceLoadParameter
-    auto& mainFrame = frame->mainFrame();
+
+    auto localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
+    ASSERT(localFrame);
```
null check.

achristensen07 commented on Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
```diff
@@ -132,7 +132,9 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
     if (auto* frame = m_document ? m_document->frame() : nullptr) {
-        if (auto* mainFrameDocumentLoader = frame->mainFrame().document() ? frame->mainFrame().document()->loader() : nullptr) {
+        auto* mainFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
+        ASSERT(mainFrame);
```
null check.

achristensen07 commented on Source/WebCore/loader/cache/CachedResourceLoader.cpp:
```diff
@@ -1051,10 +1051,13 @@ ResourceErrorOr<CachedResourceHandle<CachedResource>> CachedResourceLoader::requ
-            else if (auto* topDocument = frame.mainFrame().document())
-                sameOriginRequest = topDocument->securityOrigin().isSameSchemeHostPort(requestedOrigin.get());
+            else if (localMainFrame) {
+                if (auto* topDocument = localMainFrame->document())
```
nit: This can be a little more concise:
else if (auto* topDocument = localMainFrame ? localMainFrame->document() : nullptr)

achristensen07 commented on Source/WebCore/page/AutoscrollController.cpp:
```diff
@@ -114,8 +114,9 @@ void AutoscrollController::stopAutoscrollTimer(bool rendererIsBeingDestroyed)
-    if (frame && !frame->isMainFrame())
-        frame->mainFrame().eventHandler().didPanScrollStop();
+    auto* localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
+    if (frame && !frame->isMainFrame() && localFrame)
```
if (localFrame && !localFrame->isMainFrame())

achristensen07 commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -1209,10 +1209,12 @@ HitTestResult EventHandler::hitTestResultAtPoint(const LayoutPoint& point, Optio
-        if (RefPtr frameView = m_frame.view(), mainView = mainFrame->view(); frameView && mainView) {
-            IntPoint mainFramePoint = mainView->rootViewToContents(frameView->contentsToRootView(roundedIntPoint(point)));
-            return mainFrame->eventHandler().hitTestResultAtPoint(mainFramePoint, hitType);
+        if (auto* localFrame = dynamicDowncast<LocalFrame>(m_frame.mainFrame())) {
```
if (RefPtr mainFrame = ...) {
That way we don't need a separate line where we declare a Ref.

chirags27 committed:
Make mainFrame() return an AbstractFrame
https://bugs.webkit.org/show_bug.cgi?id=252186
rdar://105412703

Reviewed by Alex Christensen.

In preparation for website isolation, lets make Frame::mainFrame()
return an AbstractFrame. Going forward, the callers of this method will
need to decide if they are dealing with a LocalFrame or not, and take
action accordingly.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::mainFrame const):
* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
(WebCore::ContentExtensions::ContentExtensionsBackend::processContentRuleListsForLoad):
* Source/WebCore/css/query/MediaQueryFeatures.cpp:
(WebCore::MQ::Features::color):
(WebCore::MQ::Features::colorGamut):
(WebCore::MQ::Features::deviceAspectRatio):
(WebCore::MQ::Features::deviceHeight):
(WebCore::MQ::Features::deviceWidth):
(WebCore::MQ::Features::dynamicRange):
(WebCore::MQ::Features::monochrome):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::resolveStyle):
(WebCore::Document::topDocument const):
(WebCore::computeIntersectionState):
* Source/WebCore/dom/DocumentMarkerController.cpp:
(WebCore::updateMainFrameLayoutIfNeeded):
* Source/WebCore/dom/EventDispatcher.cpp:
(WebCore::shouldSuppressEventDispatchInDOM):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::canPaste const):
(WebCore::Editor::shouldInsertText const):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::allowExecutionWhenDisabledPaste):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::mainDocumentRegistrableDomainForPCM const):
(WebCore::HTMLAnchorElement::parsePrivateClickMeasurement const):
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::isElementMainContentForPurposesOfAutoplay):
(WebCore::isElementRectMostlyInMainFrame):
(WebCore::isElementLargeRelativeToMainFrame):
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::didPaint):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::disallowWebArchive const):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::userAgent const):
(WebCore::FrameLoader::navigatorPlatform const):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
(WebCore::FrameLoader::loadResourceSynchronously):
(WebCore::FrameLoader::continueLoadAfterNavigationPolicy):
(WebCore::FrameLoader::loadDifferentDocumentItem):
(WebCore::FrameLoader::loadProgressingStatusChanged):
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::FrameLoader::HistoryController::updateForCommit):
(WebCore::FrameLoader::HistoryController::updateForSameDocumentNavigation):
(WebCore::FrameLoader::HistoryController::updateBackForwardListClippedAtTarget):
(WebCore::FrameLoader::HistoryController::pushState):
* Source/WebCore/loader/NavigationDisabler.h:
(WebCore::NavigationDisabler::NavigationDisabler):
(WebCore::NavigationDisabler::~NavigationDisabler):
(WebCore::NavigationDisabler::isNavigationAllowed):
* Source/WebCore/loader/cache/CachedResource.cpp:
(WebCore::CachedResource::load):
* Source/WebCore/loader/cache/CachedResource.h:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::CachedResourceLoader::requestResource):
* Source/WebCore/page/AutoscrollController.cpp:
(WebCore::AutoscrollController::stopAutoscrollTimer):
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::outerHeight const):
(WebCore::DOMWindow::outerWidth const):
(WebCore::DOMWindow::isSameSecurityOriginAsMainFrame const):
(WebCore::DOMWindow::open):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::hitTestResultAtPoint const):
(WebCore::EventHandler::selectCursor):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::internalKeyEvent):
* Source/WebCore/page/Frame.h:
(WebCore::Frame::mainFrame const):
* Source/WebCore/page/FrameTree.cpp:
(WebCore::isFrameFamiliarWith):
(WebCore::FrameTree::find const):
(WebCore::FrameTree::traverseNext const):
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::axObjectCache const):
* Source/WebCore/page/InteractionRegion.cpp:
(WebCore::interactionRegionForRenderedRegion):
* Source/WebCore/page/IntersectionObserver.cpp:
(WebCore::IntersectionObserver::IntersectionObserver):
* Source/WebCore/page/Screen.cpp:
(WebCore::isLoadingInHeadlessMode):
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::mainDocumentLoader):
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
(WebCore::ScrollingCoordinator::updateSynchronousScrollingReasons):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::scrollBySimulatingWheelEvent):
* Source/WebCore/testing/Internals.mm:
(WebCore::Internals::rangeForDictionaryLookupAtLocation):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):
(WebKit::WebAutomationSessionProxy::takeScreenshot):
(WebKit::WebAutomationSessionProxy::snapshotRectForScreenshot):
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::WebLoaderStrategy::loadResource):
(WebKit::addParametersShared):
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect):
* Source/WebKit/WebProcess/WebCoreSupport/ShareableBitmapUtilities.cpp:
(WebKit::createShareableBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp:
(WebKit::WebResourceLoadObserver::logSubresourceLoading):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::convertDragImageToBitmap):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::isTopFrameNavigatingToAppBoundDomain const):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::rangeForPointInRootViewCoordinates):
* Source/WebKitLegacy/mac/WebView/WebImmediateActionController.mm:
(-[WebImmediateActionController performHitTestAtPoint:]):
* Source/WebKitLegacy/win/WebCoreSupport/AcceleratedCompositingContext.cpp:
(AcceleratedCompositingContext::flushPendingLayerChanges):
* Source/WebKitLegacy/win/WebCoreSupport/WebDragClient.cpp:
(WebDragClient::dragSourceActionMaskForPoint):

Canonical link: https://commits.webkit.org/260414@main
